File: week3/day1.js
const nameInput = document.getElementById("nameInput");
const jobInput = document.getElementById("jobInput");
const sendBtn = document.getElementById("sendBtn");
const result = document.getElementById("result");

sendBtn.addEventListener("click",async function(){
    const name = nameInput.value
    const job = jobInput.value

    if(!name||!job){
        result.textContent = "名前・職業を入力してね！";
        return;
    }

    const response = await fetch ("http://127.0.0.1:8000/profile",{
        method: "POST",
        headers: {"Content-Type":"application/json"},
        body: JSON.stringify({name:name,job:job}),
    });

    const data = await response.json();
    result.textContent = data.message;
});

File: week3/day1.py
from fastapi import FastAPI
from pydantic import BaseModel
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_methods=["*"],
    allow_headers=["*"],
)

class ProfileData(BaseModel):
    name: str
    job: str

@app.post("/profile")
def make_profile(data: ProfileData):
    return{"message":f"{data.name}さんは{data.job}です！"}

